11-if/questions/4-error-handling-edit.md

```diff
@@ -0,0 +1,185 @@
+(A) I don't know. <br>
+(B) To control the program's execution flow. <br>
+(C) To make the program safe from dangerous malware. <br>
+(D) Because many things can go with the program. <br>
+
+---
+
+**Solution**: (D) <br>
+Many things can potentially go wrong when running a program, so catching an error can help you write code in the program to prepare for where something cause problems.
+
+---
+
+## 2. What is a nil value? ##
+(A) The dark matter that rules the universe. <br>
+(B) It's a zero-value for pointers or pointer-based types, meaning the value has not been initialized. <br>
+(C) It's holds the same value as an empty strings (`""` == nil) <br>
+
+---
+
+**Solution**: (B) <br>
+
+---
+
+## 3. What is an error value? ##
+(A) It stores the error details. <br>
+(B) A global variable that stores the error status. <br>
+(C) A global constant which stores the error status. <br>
+
+---
+
+**Solution**: (A) <br>
+In Go, errors are given values, and therefore can be stored into a variable. 
+<br>
+There are no global variables in Go -- only package-level variables.
+
+---
+
+## 4. How does Go handle errors? ##
+(A) Using a throw/catch statement <br>
+(B) Using a simple if statement with nil comparison  <br>
+(C) Using a mechanism called tagging <br>
+
+---
+
+**Solution**: (B) <br>
+There are not throw/catch statements in Go. <br>
+Unlike Java, C#, and other languages, Go is very explicit about error handling.
+
+---
+
+## 5. When should you handle errors? ##
+(A) After the `main()` function. <br>
+(B) Before calling any function. <br>
+(C) Immediately after calling a function that returns an error value. <br>
+
+---
+
+**Solution**: (C) <br>
+When calling the function, it is a good idea initialize and assign an explicit variable to the error value returned from a function and use and if statement against the nil value to handle the errors.
+
+---
+
+## 6. Which of the following functions should you consider handling errors for?  ##
+```go
+func Read() error {...}
+func Write() error {...}
+func String() string {...}
+func Reset() {...}
+```
+(A) `Read()` and `Write()` <br>
+(B) `String()` and `Reset()` <br>
+(C) `Read()`, `Write()`, and `Reset()` <br>
+(D) None of the functions need error handling <br>
+(E) All of the functions need error handling <br>
+
+---
+
+**Solution**: (B) <br>
+The return `Read()` and `Write()` functions are the only functions that have an return value that has and error type. <br>
+`String()` and `Reset()` don't return any error values so checking for them may not be necessary. <br>
+
+---
+
+## 7. What does it mean when a function returns a `nil` error value? ##
+(A) It means the function call failed. <br>
+(B) It means the function call was successful. <br>
+(C) It means the function call is currently in an indeterministic state. <br>
+
+---
+
+**Solution**: (B) <br>
+A `nil` error value is a good thing meaning, that the code can continue executing while there is more code to execute.
```

Comment:
A `nil` error value is a good thing,<-- meaning that the code can continue executing while there is more code to execute.

_Moved the comma._